action/governance/finalizeProposal.go

```diff
@@ -0,0 +1,211 @@
+		}
+		return false, result
+	}
+	if voteResult == governance.VOTE_RESULT_PASSED {
+		proposalDistribution := ctx.ProposalMasterStore.Proposal.GetOptionsByType(proposal.Type).PassedFundDistribution
+		err := distributeFunds(ctx, proposal, &proposalDistribution)
+		if err != nil {
+			result := action.Response{
+				Events: action.GetEvent(passedProposal.Tags(), "finalize_proposal_passed_distribution_failed"),
+			}
+			return false, result
+		}
+	}
+	if voteResult == governance.VOTE_RESULT_FAILED {
+		proposalDistribution := ctx.ProposalMasterStore.Proposal.GetOptionsByType(proposal.Type).FailedFundDistribution
+		err := distributeFunds(ctx, proposal, &proposalDistribution)
+		if err != nil {
+			result := action.Response{
+				Events: action.GetEvent(passedProposal.Tags(), "finalize_proposal_failed_distribution_failed"),
+			}
+			return false, result
+		}
+	}
+	result := action.Response{
+		Events: action.GetEvent(passedProposal.Tags(), "finalize_proposal_success"),
+	}
+	return true, result
+}
+
+func distributeFunds(ctx *action.Context, proposal *governance.Proposal, proposalDistribution *governance.ProposalFundDistribution) error {
+	// Required Perimeters for Fund Distribution
+	//proposalDistribution := ctx.ProposalMasterStore.Proposal.GetOptionsByType(proposal.Type).PassedFundDistribution
+	totalFunding := governance.GetCurrentFunds(proposal.ProposalID, ctx.ProposalMasterStore.ProposalFund).Float()
+	c, ok := ctx.Currencies.GetCurrencyByName("OLT")
+	if !ok {
+		return errors.New("fund_proposal_olt_unavailable")
+	}
+	//Fund Distribution
+	//Validators
+	validatorList, err := ctx.Validators.GetValidatorSet()
+	if err != nil {
+		return err
+	}
+	validatorEarningOLT := getPercentageCoin(c, totalFunding, proposalDistribution.Validators).Divide(len(validatorList))
+	//TODO:Instead of break continue sending to rest of the validators ,and report failed Validator
+	for _, v := range validatorList {
+		err = ctx.Balances.AddToAddress(v.Address, validatorEarningOLT)
+		if err != nil {
+			return errors.Wrap(err, fmt.Sprintf("Unable to send funds to Validator :%s", v.Name))
+		}
+	}
+
+	//Fee Pool
+	err = ctx.FeePool.AddToPool(getPercentageCoin(c, totalFunding, proposalDistribution.FeePool))
+	if err != nil {
+		return errors.Wrap(err, "Failed in adding to feepool")
+	}
+	//Reward for Proposer
+	err = ctx.Balances.AddToAddress(proposal.Proposer, getPercentageCoin(c, totalFunding, proposalDistribution.ProposerReward))
+	if err != nil {
+		return errors.Wrap(err, "Failed in rewarding Proposer")
+	}
+	//Bounty Program
+	bountyAddress := action.Address(ctx.ProposalMasterStore.Proposal.GetOptions().BountyProgramAddr)
+	err = ctx.Balances.AddToAddress(bountyAddress, getPercentageCoin(c, totalFunding, proposalDistribution.BountyPool))
+	if err != nil {
+		return errors.Wrap(err, "Failed in adding to Adding to bounty program")
+	}
+	//Burn
+	burnAmount := getPercentageCoin(c, totalFunding, proposalDistribution.Burn)
```

Comment:
Are we doing something for burn here?

Reply:
> Are we doing something for burn here?

I accidentally included this source file in my branch and I'll exclude it. Tanmay is still working on this Tx and I'll let him know your comments so he can take this into consideration.